compat/cpu.h

```diff
@@ -9,23 +9,36 @@
 #ifndef SQUID_COMPAT_CPU_H
 #define SQUID_COMPAT_CPU_H
 
-#if HAVE_ERRNO_H
-#include <errno.h> /* for ENOTSUP */
-#endif
 #if HAVE_SCHED_H
 #include <sched.h>
 #endif
 
-#if !HAVE_CPU_AFFINITY
-/* failing replacements to minimize the number of if-HAVE_CPU_AFFINITYs */
-#if !HAVE_CPU_SET_T
+#if !HAVE_SCHED_SETAFFINITY
+
+#if HAVE_STDBOOL_H
+#include <stdbool.h>
+#endif
+#if HAVE_SYS_PARAM_H
+#include <sys/param.h>
+#endif
+#if HAVE_SYS_CPUSET_H
+#include <sys/cpuset.h>
+#if HAVE_CPUSET_T
+typedef cpuset_t cpu_set_t;
+#endif
+#define HAVE_CPU_AFFINITY 1
```

Comment:
We should not reset the value defined/controlled by our own ./configure. AFAICT, either ./configure should not set HAVE_CPU_AFFINITY or this code should not set HAVE_CPU_AFFINITY.

```
configure.ac:if test "x$ac_cv_func_sched_getaffinity" = "xyes" -a "x$ac_cv_func_sched_setaffinity" = "xyes" ; then
configure.ac:  AC_DEFINE(HAVE_CPU_AFFINITY,1,[Support setting CPU affinity for workers])
configure.ac:fi
```